torch/csrc/jit/node_hashing.cpp

```diff
@@ -100,6 +100,21 @@ bool ivaluesEqual(const IValue& a1, const IValue& a2) {
     at::ArrayRef<IValue> a2_elem = a2.toTuple()->elements();
     return attributesEqual(a1_elem, a2_elem);
   }
+  if (a1.isGenericDict()) {
+    auto a1_dict = a1.toGenericDict();
+    auto a2_dict = a2.toGenericDict();
+    if (a1_dict.size() != a2_dict.size()) {
+      return false;
+    }
+
+    for (const auto& entry : a1_dict) {
```

Comment:
We're not preserving key order, but that's probably fine. Maybe add comment.
`x = {}; x[0] = 1; x[1] = 2; y = {}; y[1] = 2; y[0] = 1; assert x == y`

Reply:
yup, the insertion order doesn't matter for equality. I'll add a comment.

Reply:
eh, actually i don't know if it's fine. it might break users code, and it's pretty easy to construct an example in which it does.

Reply:
sorry to flip flop here